### PR TITLE
Add support for appcheck

### PR DESCRIPTION
**Pull request by rlazo**

### CODE REVIEW

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt [hunk 1]
```diff
@@ -79,7 +86,27 @@ internal constructor(
...
+      object : HeaderProvider {
+        override val timeout: Duration
+          get() = 10.seconds
+
+        override suspend fun generateHeaders(): Map<String, String> {
+          if (appCheckTokenProvider == null) {
+            Log.w(TAG, "AppCheck not registered, skipping")
+            return emptyMap()
+          }
+          val mutableHeaderMap: MutableMap<String, String> = mutableMapOf()
```
Comment (daymxn) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt:99]: Either of these is completely fine, just wanted to make sure you know that you can also do this:

```suggestion
          val mutableHeaderMap = mutableMapOf<String, String>()
```

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt [hunk 2]
```diff
@@ -79,7 +86,27 @@ internal constructor(
...
+
+          val token = appCheckTokenProvider.getToken(false).await()
+          if (token.error != null) {
+            Log.w(TAG, "Error obtaining appcheck token", token.error)
+          }
+          mutableHeaderMap.put("X-Firebase-AppCheck", token.token)
```
Comment (daymxn) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt:105]: Will `token` be present if `token.error` is? Shouldn't you have a return in the above block?

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt [hunk 3]
```diff
@@ -79,7 +86,27 @@ internal constructor(
...
+
+          return mutableHeaderMap
```
Comment (daymxn) [inline firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt:107]: There's no need for a mutable map here fwiw:

```suggestion
          val token = appCheckTokenProvider.getToken(false).await()
       
          if (token.error != null) {
            Log.w(TAG, "Error obtaining appcheck token", token.error)
            return emptyMap()
          }
          
          return mapOf("X-Firebase-AppCheck" to token.token)
```

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt (hunk 2 above)
Comment (rlazo) [reply firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt:105]: Yeah, that's one peculiarity of appcheck, token will be present even if there's an error. 

https://github.com/firebase/firebase-android-sdk/blob/a7edffefa9cd4f45b5921db3fd2003101c38e2ef/appcheck/firebase-appcheck-interop/src/main/java/com/google/firebase/appcheck/AppCheckTokenResult.java#L21

File: firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt (hunk 3 above)
Comment (rlazo) [reply firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt]: Thanks! that's a left over from previous testing. Fixed

**Pull request merged**